mirrord/layer/src/tcp_steal/httpv2.rs

```diff
@@ -0,0 +1,170 @@
+use std::{future::Future, net::SocketAddr};
```

Comment:
Hmm I'm exploring the trait approach to try and unify this code.

- Changes:

1. [`connection.rs`](https://github.com/meowjesty/mirrord/blob/issue/922/http2-duplicate/mirrord/layer/src/tcp_steal/connection.rs#L25): try to implement the most shared logic in the traits;
2. [`httpv1.rs`](https://github.com/meowjesty/mirrord/blob/issue/922/http2-duplicate/mirrord/layer/src/tcp_steal/httpv1.rs#L47): removed individual implementations in favor of trait impl;
3. [`httpv2.rs`](https://github.com/meowjesty/mirrord/blob/issue/922/http2-duplicate/mirrord/layer/src/tcp_steal/httpv2.rs#L39) same as httpv1.

To do this we need to enable `async_fn_in_traits`, otherwise we would be boxing a bunch of stuff.

Reply:
Looks a lot better IMO, @t4lz @aviramha wdyt?